Remotion/Web/Development.WebTesting/Accessibility/AccessibilityAnalyzer.cs

```diff
@@ -241,6 +243,35 @@ private bool IsEmptyFrame ()
       return !WebDriver.FindElements (By.XPath ("/html/body/*")).Any();
     }
 
+    private void WaitUntilFrameIsVisible (int timeout = 3000)
+    {
+      var webDriverWait = new WebDriverWait (WebDriver, TimeSpan.FromMilliseconds(timeout))
+                          {
+                              PollingInterval = TimeSpan.FromMilliseconds (timeout / 10)
+                          };
+
+      try
+      {
+        webDriverWait.Until (FrameLoaded);
+      }
+      finally
+      {
+        WebDriver.SwitchTo().DefaultContent();
+      }
+    }
+
+    /// <summary>
+    /// Sometimes, due to the delayed generation of the compiled aspx file, the content of the DatePicker iframe is not fully loaded when the iframe is rendered.
+    /// </summary>
+    private bool FrameLoaded (IWebDriver driver)
+    {
+      driver.SwitchTo().DefaultContent();
+      var iframe = driver.FindElement(By.TagName("iframe"));
+      driver.SwitchTo().Frame(iframe);
+
+      return driver.FindElements(By.TagName("body")).Count > 0;
+    }
+
```

Comment:
@iam-mholle 
I don't really know how to implement this change such that it fits nicely into the existing infrastructure. That's why I have pushed it like this for now. I am not sure if this wait mechanism is "trivial enough" to warrant its co-existence alongside with `ScreenshotBocDateTimeValuePickerExtensions` (https://github.com/re-motion/Trunk-SVN2Git/pull/21/files).

I could also imagine creating a corresponding `WebDriverExtensions` class with a slightly more general version of this wait mechanism that is called accordingly from `ScreenshotBocDateTimeValuePickerExtensions` (so as not to cause invocation changes downstream) and `AccessibilityAnalyzer`.

Anyways, I am eager to learn your opinion on this.

Reply:
I'll give @iam-mholle first pick at this but I'd like to add my 2c to it after he's done.